pilot/pkg/features/pilot.go

```diff
@@ -295,8 +295,8 @@ var (
 			"By default, this is false, and virtualService with delegate will be ignored",
 	).Get()
 
-	ClusterName = env.RegisterStringVar("CLUSTER_ID", "Kubernetes",
-		"Defines the cluster and service registry that this Istiod instance is belongs to").Get()
+	ClusterName = env.RegisterStringVar("CLUSTER_ID", "default-local-istiod-cluster",
```

Comment:
I don't like Kuberenetes as 'default' - but the default-local-istiod-cluster is not better.  And the problem was not the default, but the fact that the 'local' cluster didn't know its own name. 

How about:
- leave it as it was for single-cluster ( since the new default value is not much better )
- add an option for MC so each cluster knows its name. 

I would go one step further: if 'clusterName' is defined in mesh config, then that istiod install is expected to be multi-cluster enabled. If 'networkName' is set in mesh config - that install is assumed to be multi-network enabled. 

This way we have a deterministic and easy way to know that MC/MN are enabled (now it's some mix of values.yaml/global.yaml/guessing). And user doesn't have to type multiple times. 

Since values.yaml now includes mesh config - the templates can be adjusted easily.

Reply:
> Since values.yaml now includes mesh config - the templates can be adjusted easily.

I'm pretty sure clusterName and network aren't set in meshConfig; they're in Global. They properly get set via env variables, so I'm not sure what changes would be needed in the templates. 

> leave it as it was for single-cluster ( since the new default value is not much better )

I'd really like to use anything that isn't a valid value for "ProviderID" so we aren't confusing identifiers for instances of registries with the type of registry. 

> add an option for MC so each cluster knows its name.

In multicluster scenarios it seems like the cluster name is being set just fine.